rs/messaging/src/routing/stream_handler/tests.rs

```diff
@@ -689,1112 +552,1189 @@ fn induct_loopback_stream_with_memory_limit_impl(
 #[test]
 fn induct_stream_slices_reject_response_from_old_host_subnet_is_accepted() {
-    with_test_replica_logger(|log| {
-        let (stream_handler, mut initial_state, metrics_registry) = new_fixture(&log);
-
-        // Local canister with one reservation for two incoming responses.
-        let mut initial_canister_state = new_canister_state(
-            *LOCAL_CANISTER,
-            user_test_id(24).get(),
-            *INITIAL_CYCLES,
-            NumSeconds::from(100_000),
-        );
-        make_input_queue_reservations(&mut initial_canister_state, 2, *REMOTE_CANISTER);
-        initial_state.put_canister_state(initial_canister_state);
-
-        // A stream with 3 messages in it...
-        let mut stream = generate_loopback_stream(StreamConfig {
+    with_test_setup(
+        // A stream slice with 3 messages in it...
+        btreemap![],
+        btreemap![CANISTER_MIGRATION_SUBNET => StreamSliceConfig {
             messages_begin: 0,
-            message_count: 0,
-            signals_end: 0,
-            reject_signals: None,
-        });
-        // ...a reject response for a request from `LOCAL_CANISTER` to `REMOTE_CANISTER`...
-        let reject_response = generate_reject_response_for(
-            RejectReason::QueueFull,
-            &test_request(*LOCAL_CANISTER, *REMOTE_CANISTER),
-        );
-        stream.push(reject_response.clone());
-        // ...a normal response from `REMOTE_CANISTER` to `LOCAL_CANISTER`...
-        stream.push(test_response(*REMOTE_CANISTER, *LOCAL_CANISTER).into());
-        // ...and a request from `REMOTE_CANISTER` to `LOCAL_CANISTER`...
-        stream.push(test_request(*REMOTE_CANISTER, *LOCAL_CANISTER).into());
-
-        // ...turned into a stream slice.
-        let stream_slice: StreamSlice = stream.into();
-
-        // Mark `LOCAL_CANISTER` as having migrated from `CANISTER_MIGRATION_SUBNET`
-        // to `LOCAL_SUBNET`.
-        initial_state = simulate_canister_migration(
-            initial_state,
-            *LOCAL_CANISTER,
-            CANISTER_MIGRATION_SUBNET,
-            LOCAL_SUBNET,
-        );
+            messages: vec![
+                // ...a reject response for a request sent to `REMOTE_CANISTER` @0...
+                RejectResponse(*REMOTE_CANISTER, *LOCAL_CANISTER, RejectReason::QueueFull),
+                // ...a normal response @1...
+                Response(*REMOTE_CANISTER, *LOCAL_CANISTER),
+                // ...a request from @2.
+                Request(*REMOTE_CANISTER, *LOCAL_CANISTER),
+            ],
+            ..StreamSliceConfig::default()
+        }],
+        |stream_handler, state, slices, metrics| {
+            // Mark `LOCAL_CANISTER` as having migrated from `CANISTER_MIGRATION_SUBNET` to `LOCAL_SUBNET`.
+            let state = simulate_canister_migration(
+                state,
+                *LOCAL_CANISTER,
+                CANISTER_MIGRATION_SUBNET,
+                LOCAL_SUBNET,
+            )
+            .unwrap();
```

Comment:
Do we ever not `unwrap()` the result of any of the `*_canister_migration()` functions? If not, I'd rather keep the `unwrap()` within the respective functions, they're merely test helpers / fixtures, no need to be pedantic about it.